feat: migrate logging `log` -> `tracing`

CommanderStorm opened the pull request:
<del>

Lets split this up into two PRs
- one changing the interface, tests and dependencys
- one adding additional configuration options to martin AND documenting them

Alternatively, we can keep this PR open and I am going to get back to this sometime on the weekend

</del>

Resolves #836


> we should continue using the [RUST_LOG](https://docs.rs/env_logger/latest/env_logger/#enabling-logging) env var for all the configuration, with the possible addition of CLI and Config params

- [x] `RUST_LOG`
- [x] custom env variables `MARTIN_LOG_FORMAT`, `MARTIN_LOG_LEVEL`, ...
- [x] deduplicating code
- [x] CLI
- [x] Config
- [x] Writing docs (🦖)

CommanderStorm committed:
migrate the base of the tracing implementation

CommanderStorm committed:
replace all log with tracing calls

CommanderStorm committed:
fix testcases

CommanderStorm committed:
Merge branch 'main' into logging

# Conflicts:
#	martin/src/cog/source.rs

CommanderStorm committed:
fix issue with the last merge

nyurik commented on martin/src/bin/martin-cp.rs:
```diff
@@ -442,6 +441,47 @@ async fn main() {
+                .with_default_directive(tracing::Level::INFO.into())
+                .from_env_lossy(),
+        )
+        .with((log_format == LogFormat::Compact).then(|| Layer::default().compact()))
```
oh come on, this should be a clean `match` statement  :)

CommanderStorm replied:
That would work only if this would not so in depth generics.
If I were to match on this, I would likely end up with a dyn-box, duplicate the usage below or similar.

Imo, the current version is simpler..

nyurik commented on martin/src/bin/martin-cp.rs:
```diff
@@ -442,6 +441,47 @@ async fn main() {
     }
 }
 
+fn setup_logging() {
```
this is a dup - lets not copy paste large functions like this if we can share the code

nyurik reviewed (commented):
needs a few cleanups, but this might be ok... We should probably run some perf tests (we have that using oha in the justfile iirc) - to see if there are any significant slowdowns due to the switch

nyurik committed:
Merge branch 'main' into logging

CommanderStorm committed:
remove request logging at INFO level

CommanderStorm committed:
Revert "remove request logging at INFO level"

This reverts commit bd0febac7f6acba5948fa460c9f93175cbe686bc.

CommanderStorm committed:
remove the second request logger

CommanderStorm committed:
add an per-thread intrest cache for tracing_log

CommanderStorm committed:
move obeservaibility to their own crate

CommanderStorm committed:
Merge branch 'main' into logging

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

CommanderStorm committed:
clean up and better document MartinObservability

CommanderStorm commented on martin-observability-utils/src/lib.rs:
```diff
@@ -0,0 +1,129 @@
+#![doc = include_str!("../README.md")]
+
+#[derive(Default)]
+pub struct MartinObservability {
```
Bit unsure about the API/naming here..

I tried to make it less "magic", but unsure if this is the best way to handle it.

CommanderStorm committed:
remove unused dependencys

CommanderStorm committed:
implemnt observability parsing

CommanderStorm committed:
added testcases for observability tooling

# Conflicts:
#	Cargo.lock
#	martin-observability-utils/Cargo.toml
#	martin-observability-utils/src/lib.rs

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

CommanderStorm committed:
fixed the default log filter being slightly different

CommanderStorm committed:
fix the mbtiles tests not having tracing support

CommanderStorm committed:
remove accidental change

CommanderStorm committed:
improve CLI rendering

CommanderStorm committed:
update the docs

CommanderStorm committed:
adapt to required new usage according to docs I just wrote, sigh...

CommanderStorm committed:
make sure that the cli of martin, martin-cp and mbtiles does format log formatting and log levels the same way

CommanderStorm committed:
add support for config file parsing and add an explainaition why these fields are not used

CommanderStorm committed:
remove terminal escape characters from the log before comparing the output

CommanderStorm committed:
Merge branch 'main' into logging

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

CommanderStorm committed:
fix `mbtiles` logging at a different filter level as before

CommanderStorm commented on martin-observability-utils/src/lib.rs:
```diff
@@ -0,0 +1,292 @@
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub enum LogFormatOptions {
```
Currently the `bare` formatting option (added to support mbtiles as before) looks like this:

![image](https://github.com/user-attachments/assets/7cb9f5a7-9bae-4507-b47e-07e21ccf907b)


We could remove the target via adding this to the layer
`.map_event_format(|e| e.with_target(false))`

=> should mbtiles remain "as-is" or gain targets?

CommanderStorm committed:
add bare as a format option

CommanderStorm committed:
change from removing terminal escape sequences to not emitting them in the first place

CommanderStorm committed:
bless terminal outputs

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

nyurik committed:
chore: minor log and metadata cleanup (#1737)

CommanderStorm committed:
chore: improve CLI line breaks and markdown-handling (#1736)

During #1720 I noticed that there are a few places in the CLI where a
few more line breaks and handling markdown better could help
readability.

Yes, this adds the `unstable-markdown` feature of clap, but given that
removal would only impact the CLI-Formatting, I think here the risk is
okay.
Tracking issue:
- https://github.com/clap-rs/clap/issues/5900

Hidden rationale:
I have one place in #1720 where making it readable without markdown
formatting is nearly impossible

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump axios from 1.7.7 to 1.8.2 in /demo/frontend (#1727)

Bumps [axios](https://github.com/axios/axios) from 1.7.7 to 1.8.2.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/axios/axios/releases">axios's
releases</a>.</em></p>
<blockquote>
<h2>Release v1.8.2</h2>
<h2>Release notes:</h2>
<h3>Bug Fixes</h3>
<ul>
<li><strong>http-adapter:</strong> add allowAbsoluteUrls to path
building (<a
href="https://redirect.github.com/axios/axios/issues/6810">#6810</a>)
(<a
href="https://github.com/axios/axios/commit/fb8eec214ce7744b5ca787f2c3b8339b2f54b00f">fb8eec2</a>)</li>
</ul>
<h3>Contributors to this release</h3>
<ul>
<li><!-- raw HTML omitted --> <a href="https://github.com/lexcorp16"
title="+1/-1 ([#6810](https://github.com/axios/axios/issues/6810)
)">Fasoro-Joseph Alexander</a></li>
</ul>
<h2>Release v1.8.1</h2>
<h2>Release notes:</h2>
<h3>Bug Fixes</h3>
<ul>
<li><strong>utils:</strong> move <code>generateString</code> to platform
utils to avoid importing crypto module into client builds; (<a
href="https://redirect.github.com/axios/axios/issues/6789">#6789</a>)
(<a
href="https://github.com/axios/axios/commit/36a5a620bec0b181451927f13ac85b9888b86cec">36a5a62</a>)</li>
</ul>
<h3>Contributors to this release</h3>
<ul>
<li><!-- raw HTML omitted --> <a
href="https://github.com/DigitalBrainJS" title="+51/-47
([#6789](https://github.com/axios/axios/issues/6789) )">Dmitriy
Mozgovoy</a></li>
</ul>
<h2>Release v1.8.0</h2>
<h2>Release notes:</h2>
<h3>Bug Fixes</h3>
<ul>
<li><strong>examples:</strong> application crashed when navigating
examples in browser (<a
href="https://redirect.github.com/axios/axios/issues/5938">#5938</a>)
(<a
href="https://github.com/axios/axios/commit/1260ded634ec101dd5ed05d3b70f8e8f899dba6c">1260ded</a>)</li>
<li>missing word in SUPPORT_QUESTION.yml (<a
href="https://redirect.github.com/axios/axios/issues/6757">#6757</a>)
(<a
href="https://github.com/axios/axios/commit/1f890b13f2c25a016f3c84ae78efb769f244133e">1f890b1</a>)</li>
<li><strong>utils:</strong> replace getRandomValues with crypto module
(<a
href="https://redirect.github.com/axios/axios/issues/6788">#6788</a>)
(<a
href="https://github.com/axios/axios/commit/23a25af0688d1db2c396deb09229d2271cc24f6c">23a25af</a>)</li>
</ul>
<h3>Features</h3>
<ul>
<li>Add config for ignoring absolute URLs (<a
href="https://redirect.github.com/axios/axios/issues/5902">#5902</a>)
(<a
href="https://redirect.github.com/axios/axios/issues/6192">#6192</a>)
(<a
href="https://github.com/axios/axios/commit/32c7bcc0f233285ba27dec73a4b1e81fb7a219b3">32c7bcc</a>)</li>
</ul>
<h3>Reverts</h3>
<ul>
<li>Revert &quot;chore: expose fromDataToStream to be consumable (<a
href="https://redirect.github.com/axios/axios/issues/6731">#6731</a>)&quot;
(<a
href="https://redirect.github.com/axios/axios/issues/6732">#6732</a>)
(<a
href="https://github.com/axios/axios/commit/1317261125e9c419fe9f126867f64d28f9c1efda">1317261</a>),
closes <a
href="https://redirect.github.com/axios/axios/issues/6731">#6731</a> <a
href="https://redirect.github.com/axios/axios/issues/6732">#6732</a></li>
</ul>
<h3>BREAKING CHANGES</h3>
<ul>
<li>
<p>code relying on the above will now combine the URLs instead of prefer
request URL</p>
</li>
<li>
<p>feat: add config option for allowing absolute URLs</p>
</li>
<li>
<p>fix: add default value for allowAbsoluteUrls in buildFullPath</p>
</li>
<li>
<p>fix: typo in flow control when setting allowAbsoluteUrls</p>
</li>
</ul>
<h3>Contributors to this release</h3>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/axios/axios/blob/v1.x/CHANGELOG.md">axios's
changelog</a>.</em></p>
<blockquote>
<h2><a
href="https://github.com/axios/axios/compare/v1.8.1...v1.8.2">1.8.2</a>
(2025-03-07)</h2>
<h3>Bug Fixes</h3>
<ul>
<li><strong>http-adapter:</strong> add allowAbsoluteUrls to path
building (<a
href="https://redirect.github.com/axios/axios/issues/6810">#6810</a>)
(<a
href="https://github.com/axios/axios/commit/fb8eec214ce7744b5ca787f2c3b8339b2f54b00f">fb8eec2</a>)</li>
</ul>
<h3>Contributors to this release</h3>
<ul>
<li><!-- raw HTML omitted --> <a href="https://github.com/lexcorp16"
title="+1/-1 ([#6810](https://github.com/axios/axios/issues/6810)
)">Fasoro-Joseph Alexander</a></li>
</ul>
<h2><a
href="https://github.com/axios/axios/compare/v1.8.0...v1.8.1">1.8.1</a>
(2025-02-26)</h2>
<h3>Bug Fixes</h3>
<ul>
<li><strong>utils:</strong> move <code>generateString</code> to platform
utils to avoid importing crypto module into client builds; (<a
href="https://redirect.github.com/axios/axios/issues/6789">#6789</a>)
(<a
href="https://github.com/axios/axios/commit/36a5a620bec0b181451927f13ac85b9888b86cec">36a5a62</a>)</li>
</ul>
<h3>Contributors to this release</h3>
<ul>
<li><!-- raw HTML omitted --> <a
href="https://github.com/DigitalBrainJS" title="+51/-47
([#6789](https://github.com/axios/axios/issues/6789) )">Dmitriy
Mozgovoy</a></li>
</ul>
<h1><a
href="https://github.com/axios/axios/compare/v1.7.9...v1.8.0">1.8.0</a>
(2025-02-25)</h1>
<h3>Bug Fixes</h3>
<ul>
<li><strong>examples:</strong> application crashed when navigating
examples in browser (<a
href="https://redirect.github.com/axios/axios/issues/5938">#5938</a>)
(<a
href="https://github.com/axios/axios/commit/1260ded634ec101dd5ed05d3b70f8e8f899dba6c">1260ded</a>)</li>
<li>missing word in SUPPORT_QUESTION.yml (<a
href="https://redirect.github.com/axios/axios/issues/6757">#6757</a>)
(<a
href="https://github.com/axios/axios/commit/1f890b13f2c25a016f3c84ae78efb769f244133e">1f890b1</a>)</li>
<li><strong>utils:</strong> replace getRandomValues with crypto module
(<a
href="https://redirect.github.com/axios/axios/issues/6788">#6788</a>)
(<a
href="https://github.com/axios/axios/commit/23a25af0688d1db2c396deb09229d2271cc24f6c">23a25af</a>)</li>
</ul>
<h3>Features</h3>
<ul>
<li>Add config for ignoring absolute URLs (<a
href="https://redirect.github.com/axios/axios/issues/5902">#5902</a>)
(<a
href="https://redirect.github.com/axios/axios/issues/6192">#6192</a>)
(<a
href="https://github.com/axios/axios/commit/32c7bcc0f233285ba27dec73a4b1e81fb7a219b3">32c7bcc</a>)</li>
</ul>
<h3>Reverts</h3>
<ul>
<li>Revert &quot;chore: expose fromDataToStream to be consumable (<a
href="https://redirect.github.com/axios/axios/issues/6731">#6731</a>)&quot;
(<a
href="https://redirect.github.com/axios/axios/issues/6732">#6732</a>)
(<a
href="https://github.com/axios/axios/commit/1317261125e9c419fe9f126867f64d28f9c1efda">1317261</a>),
closes <a
href="https://redirect.github.com/axios/axios/issues/6731">#6731</a> <a
href="https://redirect.github.com/axios/axios/issues/6732">#6732</a></li>
</ul>
<h3>BREAKING CHANGES</h3>
<ul>
<li>
<p>code relying on the above will now combine the URLs instead of prefer
request URL</p>
</li>
<li>
<p>feat: add config option for allowing absolute URLs</p>
</li>
<li>
<p>fix: add default value for allowAbsoluteUrls in buildFullPath</p>
</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/axios/axios/commit/a9f7689b0c4b6d68c7f587c3aa376860da509d94"><code>a9f7689</code></a>
chore(release): v1.8.2 (<a
href="https://redirect.github.com/axios/axios/issues/6812">#6812</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/fb8eec214ce7744b5ca787f2c3b8339b2f54b00f"><code>fb8eec2</code></a>
fix(http-adapter): add allowAbsoluteUrls to path building (<a
href="https://redirect.github.com/axios/axios/issues/6810">#6810</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/98120457559e573024862e2925d56295a965ad7e"><code>9812045</code></a>
chore(sponsor): update sponsor block (<a
href="https://redirect.github.com/axios/axios/issues/6804">#6804</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/72acf759373ef4e211d5299818d19e50e08c02f8"><code>72acf75</code></a>
chore(sponsor): update sponsor block (<a
href="https://redirect.github.com/axios/axios/issues/6794">#6794</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/2e64afdff5c41e38284a6fb8312f2745072513a1"><code>2e64afd</code></a>
chore(release): v1.8.1 (<a
href="https://redirect.github.com/axios/axios/issues/6800">#6800</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/36a5a620bec0b181451927f13ac85b9888b86cec"><code>36a5a62</code></a>
fix(utils): move <code>generateString</code> to platform utils to avoid
importing crypto...</li>
<li><a
href="https://github.com/axios/axios/commit/cceb7b1e154fbf294135c93d3f91921643bbe49f"><code>cceb7b1</code></a>
chore(release): v1.8.0 (<a
href="https://redirect.github.com/axios/axios/issues/6795">#6795</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/23a25af0688d1db2c396deb09229d2271cc24f6c"><code>23a25af</code></a>
fix(utils): replace getRandomValues with crypto module (<a
href="https://redirect.github.com/axios/axios/issues/6788">#6788</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/32c7bcc0f233285ba27dec73a4b1e81fb7a219b3"><code>32c7bcc</code></a>
feat: Add config for ignoring absolute URLs (<a
href="https://redirect.github.com/axios/axios/issues/5902">#5902</a>)
(<a
href="https://redirect.github.com/axios/axios/issues/6192">#6192</a>)</li>
<li><a
href="https://github.com/axios/axios/commit/4a3e26cf65bb040b7eb4577d5fd62199b0f3d017"><code>4a3e26c</code></a>
chore(config): adjust rollup config to preserve license header to
minified Ja...</li>
<li>Additional commits viewable in <a
href="https://github.com/axios/axios/compare/v1.7.7...v1.8.2">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=axios&package-manager=npm_and_yarn&previous-version=1.7.7&new-version=1.8.2)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)
You can disable automated security fix PRs for this repo from the
[Security Alerts
page](https://github.com/maplibre/martin/network/alerts).

</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

CommanderStorm committed:
Merge branch 'main' into logging

CommanderStorm committed:
tmp-debugging

nyurik committed:
Update deps, fix new sqlx usage (#1428)

Resolves https://github.com/maplibre/martin/issues/1475

One possible concern is that i32 is now used as i64 with sqlx. This
**might** impact `tile_xxh3_64_hash` value validity, but I don't see
conclusive evidence of that. Note that the tests were incorrectly dumped
as i32, thus had to be updated, but it does not seem to affect the
actual storage or validation...

---------

Co-authored-by: Frank Elsinga <frank@elsinga.de>

CommanderStorm committed:
re-bless after testcase addition

CommanderStorm committed:
Merge branch 'main' into logging

CommanderStorm committed:
test using the `language_version` key

CommanderStorm committed:
Revert "test using the `language_version` key"

This reverts commit 462047c4eea3a9f3535982cf347f7493048bf4f7.

CommanderStorm committed:
move the tracing to its own crate

CommanderStorm committed:
Merge branch 'main' into martin-tracing

CommanderStorm committed:
fix wrong dependency definition in cargo.toml

CommanderStorm committed:
add debug logging for NO_COLOR

CommanderStorm commented on tests/test.sh:
```diff
@@ -27,6 +27,10 @@ TEST_TEMP_DIR="$(dirname "$0")/mbtiles_temp_files"
 
+# by default, martin/.. do pretty up their output for terminals with colors
+# in CI, while comparing outputs, this makes testcases less readable
+export NO_COLOR=true
```
This *SHOULD* work, but for some reason it does only locally.. no clue why..

This also just *SOMETIMES* does this.. so weird

Will have to look further into the code what is overriding this being active...

CommanderStorm committed:
Merge branch 'main' into martin-tracing

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

CommanderStorm committed:
clairify why test_log_level_env_var 'can' use std::env::set_var

CommanderStorm committed:
Merge branch 'maplibre:main' into martin-tracing

CommanderStorm committed:
Merge branch 'main' into martin-tracing

CommanderStorm committed:
cargo update

CommanderStorm committed:
cargo update

CommanderStorm committed:
update the testcases and allow dot notation for accessing strings

CommanderStorm committed:
Merge branch 'main' into logging

CommanderStorm committed:
Merge branch 'martin-tracing' into logging

CommanderStorm committed:
fix slightly bodged merge from martin-tracing leaving a second folder

CommanderStorm committed:
fix removing the wrong directory in merge-fixing

CommanderStorm committed:
fix clippy issue

CommanderStorm committed:
Merge branch 'martin-tracing' into logging

CommanderStorm committed:
fix md-fmt issue

CommanderStorm committed:
Merge branch 'main' into logging

CommanderStorm committed:
fix a few of the rebasing issues
